src/App.jsx

```diff
@@ -1,28 +1,97 @@
 import React from 'react';
 import './App.scss';
 import './styles/general.scss';
+import { getTodos } from './api/api';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
 
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
+    statusFilter: 'All',
+    titleFilter: '',
   };
 
+  componentDidMount = async() => {
+    const todos = await getTodos();
+
+    this.setState({
+      todos: todos.data.filter(todo => todo.title && todo.id && todo.userId),
+    });
+  }
+
+  handleSelect = (event) => {
+    this.setState({
+      statusFilter: event.target.value,
+    });
+  }
+
+  handleChange = (input) => {
+    this.setState({
+      titleFilter: input,
+    });
+  }
+
+  setUser = (userId) => {
+    this.setState({
+      selectedUserId: userId,
+    });
+  }
+
+  clearUser = () => {
+    this.setState({
+      selectedUserId: 0,
+    });
+  }
+
   render() {
-    const { todos, selectedUserId } = this.state;
+    const { todos, selectedUserId, statusFilter, titleFilter } = this.state;
+
+    let filteredByTitle;
+
+    if (titleFilter) {
+      filteredByTitle = todos.filter(todo => (
+        todo.title && todo.title.includes(titleFilter)
+      ));
+    } else {
+      filteredByTitle = todos;
+    }
+
+    let preparedTodos;
+
+    switch (statusFilter) {
+      case 'Completed':
+        preparedTodos = filteredByTitle.filter(todo => todo.completed);
+        break;
+      case 'Not completed':
+        preparedTodos = filteredByTitle.filter(todo => !todo.completed);
+        break;
+      default:
+        preparedTodos = filteredByTitle;
+        break;
+    }
 
     return (
       <div className="App">
         <div className="App__sidebar">
-          <TodoList todos={todos} />
+          <TodoList
+            todos={preparedTodos}
+            handleSelect={this.handleSelect}
+            handleChange={this.handleChange}
+            statusFilter={this.state.statusFilter}
+            titleFilter={this.state.titleFilter}
```

Comment:
```suggestion
            statusFilter={statusFilter}
            titleFilter={titleFilter}
```
ти в рендері зробив деструктуризацію, але все одно навіщо їх передавати в TodoList - не розумію

Reply:
трошки неуважний) Пофіксив)